packages/idyll-cli/src/pipeline/parse.js

```diff
@@ -14,102 +20,25 @@ const getFilteredAST = (ast) => {
   });
 }
 
-const getNodesByName = (name, tree) => {
-  const predicate = typeof name === 'string' ? (s) => s === name : name;
-
-  const byName = (acc, val) => {
-    if (typeof val === 'string') return acc;
-    if (predicate(val[0])) acc.push(val);
-
-    if (val[2] && typeof val[2] !== 'string') acc = val[2].reduce(byName, acc);
-
-    return acc;
-  }
-
-  return tree.reduce(
-    byName,
-    []
-  )
+exports.getComponentNodes = (ast) => {
+  const ignoreNames = ['var', 'data', 'meta', 'derived'];
+  return filterNodes(ast, s => !ignoreNames.includes(s));
```

Comment:
I believe this will return the nodes still nested, when we really want to return a flattened list